### PR TITLE
Support mv a path into an existing dir

### PR DESCRIPTION
**Pull request by singer-bin**
### What changes are proposed in this pull request?

Support mv a path into an existing dir.

### Why are the changes needed?
Support mv a path into an existing dir.

### Does this PR introduce any user facing changes?
 N/A

### CODE REVIEW

**Comment by singer-bin** [comment]
The failure of some tests is not related to my modification. @LuQQiu @dbw9580

File: shell/src/main/java/alluxio/cli/fs/command/MvCommand.java [hunk 1]
```diff
@@ -52,6 +52,9 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI srcPath = new AlluxioURI(args[0]);
     AlluxioURI dstPath = new AlluxioURI(args[1]);
+    if (mFileSystem.exists(dstPath) && mFileSystem.getStatus(dstPath).isFolder()) {
```
Comment (maobaolong) [inline shell/src/main/java/alluxio/cli/fs/command/MvCommand.java:55]: I guess you propose to do this change to keep consistent with hdfs, but shall we move this logic to the alluxio master side?

File: tests/src/test/java/alluxio/client/cli/fs/command/MvCommandIntegrationTest.java [hunk 2]
```diff
@@ -58,10 +58,10 @@ public void renameToExistingFile() throws IOException {
...
-        "Cannot rename because destination already exists. src: /testFolder1 dst: /testFolder"));
-  }
-}
+    Assert.assertEquals(0, ret);
+    Assert.assertTrue(fileExists(new AlluxioURI("/testFolder/testFolder1")));
+    Assert.assertFalse(fileExists(new AlluxioURI("/testFolder1")));
+    Assert.assertTrue(mOutput.toString().contains(
+            "Renamed /testFolder1 to /testFolder/testFolder1"));
+   }
+ }
```
Comment (maobaolong) [inline tests/src/test/java/alluxio/client/cli/fs/command/MvCommandIntegrationTest.java:67]: Please add a new line here and fix the ident issue.

Comment (singer-bin) [reply tests/src/test/java/alluxio/client/cli/fs/command/MvCommandIntegrationTest.java:67]: I will fix it right now.

File: shell/src/main/java/alluxio/cli/fs/command/MvCommand.java (hunk 1 above)
Comment (singer-bin) [reply shell/src/main/java/alluxio/cli/fs/command/MvCommand.java:55]: Yes, I want to keep ALLUXIO's `mv` behavior consistent with HDFS. Is there any benefit to moving this operation to the master side? Can you suggest how to modify it?

**Comment by LuQQiu** [comment]
Similar to fuse side mv changes https://github.com/Alluxio/alluxio/pull/15708/files
we can maintain the logics on client side for now

**Comment by LuQQiu** [comment]
@singer-bin what's the current master behavior of mv? what's the error message?
can you help check fuse behavior as well? i guess fuse mv may have similar behavior as the current master behavior of alluxio fs mv and may need to be fixed as well, thanks!

**Comment by singer-bin** [comment]
> @singer-binmv 目前的主要行为是什么？错误信息是什么？ 你能帮忙检查保险丝的行为吗？我猜 fuse mv 的行为可能与 alluxio fs mv 的当前主行为相似，并且可能也需要修复，谢谢！



> @singer-binmv 当前的主要行为是什么？错误信息是什么？ 你能帮忙保险丝的行为吗？我猜 fuse mv 的行为与 alluxio fs mv 的当前主行为相似，并且可能也需要修复，谢谢检查！

**Comment by singer-bin** [comment]
> @singer-bin what's the current master behavior of mv? what's the error message? can you help check fuse behavior as well? i guess fuse mv may have similar behavior as the current master behavior of alluxio fs mv and may need to be fixed as well, thanks!

Before adding PR:
``` 
$ bin/alluxio fs mkdir /aaa/
Successfully created directory /aaa`
$ bin/alluxio fs mkdir /bbb/
Successfully created directory /bbb
$ bin/alluxio fs copyFromLocal /home/ocdp/a.txt /bbb/
Copied file:///home/ocdp/a.txt to /bbb
$ bin/alluxio fs mv /aaa /bbb
Cannot rename because destination already exists. src: /aaa dst: /bbb
```
After adding PR:
```
$ bin/alluxio fs mv /aaa /bbb
Renamed /aaa to /bbb/aaa
$ bin/alluxio fs ls /bbb
drwxr-xr-x  alluxio           alluxio                         0   NOT_PERSISTED 09-28-2022 17:32:45:645  DIR /bbb/aaa
-rw-r--r--  alluxio           alluxio                     11090       PERSISTED 09-28-2022 17:39:15:653 100% /bbb/a.txt
```

**Comment by singer-bin** [comment]
> Similar to fuse side mv changes https://github.com/Alluxio/alluxio/pull/15708/files we can maintain the logics on client side for now

My current modification is on the client side. If I don't need to add a flag like fuse, then there seems to be no problem with the current modification.

**Comment by singer-bin** [comment]
@LuQQiu Can you add some comments, thanks a lot.

**Pull request closed without merging**